File: Problems/Hirarchical Feature Flag System/Python/repository.py
class InMemoryChannelRepository:
    def __init__(self):
        self.channels = {}

    def add(self, channel_id, channel):
        self.channels[channel_id] = channel

    def get(self, channel_id):
        return self.channels.get(channel_id)

    def delete(self, channel_id):
        if channel_id in self.channels:
            del self.channels[channel_id]

File: Problems/Hirarchical Feature Flag System/Go/repository.go
package main

type Repository interface {
	List() []*Channel
	Get(id string) *Channel
	Add(id string, channel *Channel)
	Delete(id string)
}

type ChannelRepository interface {
	Repository
}

type InMemoryChannelRepository struct {
	channels map[string]*Channel
}

func NewInMemoryChannelRepository() *InMemoryChannelRepository {
	return &InMemoryChannelRepository{
		channels: make(map[string]*Channel),
	}
}

func (r *InMemoryChannelRepository) List() []*Channel {
	var channels []*Channel

	for _, channel := range r.channels {
		channels = append(channels, channel)
	}

	return channels
}

func (r *InMemoryChannelRepository) Get(id string) *Channel {
	channel, ok := r.channels[id]
	if !ok {
		return nil
	}
	return channel
}

func (r *InMemoryChannelRepository) Add(id string, channel *Channel) {
	r.channels[id] = channel
}

func (r *InMemoryChannelRepository) Delete(id string) {
	_, ok := r.channels[id]
	if !ok {
		return
	}
	delete(r.channels, id)
}
